File: app/src/main/java/io/sparkled/AppSettings.java
package io.sparkled;

import com.beust.jcommander.Parameter;

/**
 * Settings provided to the application as command line arguments.
 */
class AppSettings {

    @Parameter(names = {"-h", "--help"}, description = "Show help", help = true)
    private boolean help = false;

    @Parameter(names = {"-rp", "--restPort"}, description = "The REST server port")
    private int restApiPort = 8080;

    @Parameter(names = {"-up", "--udpPort"}, description = "The UDP server port")
    private int udpPort = 12345;

    @Parameter(names = {"-l", "--loglevel"}, description = "Logging level threshold (DEBUG, INFO, WARN, ERROR)")
    private String logLevel = "INFO";

    public int getRestApiPort() {
        return restApiPort;
    }

    public int getUdpPort() {
        return udpPort;
    }

    public String getLogLevel() {
        return logLevel;
    }

    public boolean isHelp() {
        return help;
    }
}

File: app/src/main/kotlin/io/sparkled/AppSettings.kt
package io.sparkled

import com.beust.jcommander.Parameter

/**
 * Settings provided to the application as command line arguments.
 */
internal class AppSettings {

    @Parameter(names = ["-h", "--help"], description = "Show help", help = true)
    var isHelp = false

    @Parameter(names = ["-rp", "--restPort"], description = "The REST server port")
    var restApiPort = 8080

    @Parameter(names = ["-up", "--udpPort"], description = "The UDP server port")
    var udpPort = 12345

    @Parameter(names = ["-l", "--logLevel"], description = "Logging level threshold (DEBUG, INFO, WARN, ERROR)")
    var logLevel = "INFO"
}
